File: javascript/clock/clock.js
export class Clock {
    constructor(hours = 0, minutes = 0) {
        this.hours = hours;
        this.minutes = minutes;

        this.wrap();
    }

    toString() {
        let hh = this.hours.toString().padStart(2, '0');
        let mm = this.minutes.toString().padStart(2, '0');

        return `${hh}:${mm}`;
    }

    plus(n) {
        this.minutes += n;

        this.wrap();

        return this;
    }

    minus(n) {
        return this.plus(-n);
    }

    wrap() {
        let overflow = Math.floor(this.minutes / 60);

        this.hours += overflow;

        this.minutes %= 60;
        if (this.minutes < 0) this.minutes += 60;

        this.hours %= 24;
        if (this.hours < 0) this.hours += 24;
    }

    equals(other) {
        return this.hours == other.hours && this.minutes == other.minutes;
    }
}

File: csharp/clock/Clock.cs
using System;

public class Clock
{
    int Hours, Minutes;

    public Clock(int hours, int minutes)
    {
        Hours = hours;
        Minutes = minutes;

        Wrap();
    }

    void Wrap()
    {
        while (Minutes >= 60) {
            Minutes -= 60;
            Hours += 1;
        }

        while (Minutes < 0) {
            Minutes += 60;
            Hours -= 1;
        }

        if (Hours >= 24)
            Hours %= 24;

        while (Hours < 0)
            Hours += 24;
    }

    public Clock Add(int minutesToAdd)
    {
        return new Clock(Hours, Minutes + minutesToAdd);
    }

    public Clock Subtract(int minutesToSubtract)
    {
        return Add(-minutesToSubtract);
    }

    public override string ToString()
    {
        var hh = Hours.ToString().PadLeft(2, '0');
        var mm = Minutes.ToString().PadLeft(2, '0');

        return hh + ':' + mm;
    }
}
